### ISSUE TITLE
Issue with slow acceleration of high torque vehicles

### ISSUE THREAD

**Issue by Ono-Sendai** [jrouwe/JoltPhysics#297]
Hi,
I'm trying out the vehicle controller.
It's mostly working, but the acceleration in the forwards direction seems weird.  The vehicle struggles to move forwards very fast.
However when accelerating backwards (reversing), the vehicle accelerates very fast (and even does wheelies).

Settings are mostly the same as in the VehicleConstraintTest, except for mEngine.mMaxTorque which I set to 2000, and also my world-space up is the z-axis.

(Regarding the world-space z-axis being up, I have set 

JPH::VehicleConstraintSettings vehicle;
vehicle.mUp = JPH::Vec3(0,0,1);
vehicle.mForward = JPH::Vec3(0,1,0);

and

w->mDirection = JPH::Vec3(0,0,-1)  

for the wheel settings.)

Although this issue is most apparent with high torque, I think it also occurs for the default torque.

Please see this video capture:  https://youtu.be/Ml_64uOW4nM

In the first half of the video I'm holding down the forwards key.  You can see the acceleration is extremely slow for such a high torque.
In the second half of the video I reverse, showing the kind of high acceleration (causing a wheelie), that I would expect.

Cheers,
  Nick

**Comment by jrouwe** [comment]
I think this is due to the automatic gearbox. When you have this much torque the vehicle makes a wheelie at which point the wheels start spinning, and then because the RPM is too high, the gearbox switches to the next gear. When the vehicle falls back to the ground, the wheels slow down and the gearbox sees that the wheels rotate too slowly for the current gear so it switches back. This process repeats until the car reaches its desired velocity and because the clutch is pressed while switching gears, you see this as the car not accelerating properly.

In reverse, there's only 1 gear, so the gearbox cannot oscillate and you get proper acceleration.

I'm not an expert on automatic transmissions so I don't know what solution for this is implemented in real cars, but I guess shifting gear up is undesirable if the wheels are spinning (so that's something I'm going to need to fix).

**Comment by Ono-Sendai** [comment]
Ok thank you, yes that is probably what's happening.  I have found that using some rear wheel drive helps. (or just using a single forwards gear)

**Comment by jrouwe** [comment]
I've applied a fix for your high torque problems, see #300. Note that the issue was not that the wheels left the ground, but that the engine RPM calculations are wildly inaccurate at this high torque causing the gearbox to jitter. I've added some latency between switching gears that seems to solve most of the problems, but the engine RPM calculation is still very inaccurate, so I'm working on a better engine model.
